docs/README.md

```diff
@@ -263,24 +264,113 @@ Structure the `search-result-layout` or the `search-result-layout.customQuery`,
+
+Find below the available blocks to build your store's search results page and their existing props as well. 
+
+#### `gallery` block
+
+The `gallery` block defines how fetched items should be displayed on the store's search results page.
+
+When declared with no props, it expects as child the [`product-summary-shelf`](https://vtex.io/docs/components/content-blocks/vtex.product-summary/) and consequently the block structure inherited from it. 
+
+It is possible, however, to use its `layouts` prop to provide several layouts to the page, allowing your store to have different arrangement of items according to what best fits your users' needs.
+
+In a scenario where several layouts are provided, your store users are able to shift between them according to their needs using the `gallery-layout-switcher` block (described further below). The `gallery` will then render the component provided by the currently selected layout. 
+
+:information_source: *You can also set a default layout for your search results page in the `defaultGalleryLayout` prop from the `search-result-layout.desktop` and `search-result-layout.mobile` blocks.*
+
+| Prop name | Type                      | Description                                                    | Default value   |
+| :---------: | :---------------------: | :--------------------------------------------------------------| :-------------: |
+| `layouts`  | `object` | List of layouts used to arrange and display the items on search results page. Caution: If no value is provided, the `gallery` block must receive instead a `product-summary-shelf` block as child. | `undefined`  |
+
+- **`layouts` object:** 
+
+| Prop name   | Type     | Description                                                             | Default value   |
+| :---------: | :------: | :---------------------------------------------------------------------: | :-------------: |
+| `name`   | `string` | ![https://img.shields.io/badge/-Mandatory-red](https://img.shields.io/badge/-Mandatory-red) Layout name. This value must be unique i.e. not equal to other layout names declared in the `gallery` block. | `undefined` |
+| `component`   | `string` | ![https://img.shields.io/badge/-Mandatory-red](https://img.shields.io/badge/-Mandatory-red) Name of the [Slot](https://vtex.io/docs/concepts/slots) that will declare the block to be rendered in this layout. The same Slot name used as this prop's value must be declared afterwards as a prop (in the `gallery` block) whose value must be the desired block name. Caution: the `component`  prop's value must be equal to the Slot name responsible for declaring the desired block, thus being PascalCased, not to the desired block name itself. Check out the example below in order to understand the underlying logic behind this prop. | `undefined` |
```

Comment:
`thus being PascalCased` -> I didn't understand this part.

Reply:
I have a question: how users should know the slot name to be filled in the `component` prop? In the example below, we have `WholeLineSummary` and `ManyByLineSummary`. How should they know about these names? We do not explain this in the Slots documentation.

Reply:
`being PascalCased` means that the first letter of each word in a compound word is capitalized. Slots' names must be PascalCased by definition, if not, they won't work.

About the second comment, the Slot name is up to the users to define. They can give it any name they want, but it is required that the `component` prop have the same name as the corresponding Slot that they want to display for that layout.

In the example, we could've used any PascalCased name for the Slot, such as `EntireLineSum`, and then define a layout that receives "EntireLineSum" as the value for the component prop.